lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -478,6 +474,56 @@ static void
   lookupTypeMembers(BaseTy, PT, Consumer, CurrDC, LS, Reason);
 }
 
+/// Trigger synthesizing implicit member declarations to make them "visible".
+static void synthesizeMemberDeclsForLookup(NominalTypeDecl *NTD,
+                                           const DeclContext *DC) {
+  // Synthesize the memberwise initializer for structs or default initializer
+  // for classes.
+  if (!NTD->hasInterfaceType())
+    TypeChecker::addImplicitConstructors(NTD);
+
+  // Check all conformances to trigger the synthesized decl generation.
+  // e.g. init(rawValue:) for RawRepresentable.
+  auto resolver = NTD->getASTContext().getLazyResolver();
+  if (!resolver)
+    return;
+  for (auto Conformance : NTD->getAllConformances()) {
+    auto Proto = Conformance->getProtocol();
+    if (!Proto->isAccessibleFrom(DC))
+      continue;
+    auto NormalConformance = dyn_cast<NormalProtocolConformance>(
+        Conformance->getRootConformance());
+    if (!NormalConformance)
+      continue;
+    for (auto Member : Proto->getMembers()) {
+      if (auto *ATD = dyn_cast<AssociatedTypeDecl>(Member)) {
+        if (!NormalConformance->hasTypeWitness(ATD))
+          resolver->resolveTypeWitness(NormalConformance, ATD);
+      } else if (auto *VD = dyn_cast<ValueDecl>(Member)) {
+        if (!NormalConformance->hasWitness(VD))
+          resolver->resolveWitness(NormalConformance, VD);
```

Comment:
Can we do this all this inside lookupDeclsFromProtocolsBeingConformedTo() instead?

Reply:
I don't think so because this has to be done before `lookupTypeMembers()` which is called before `lookupDeclsFromProtocolsBeingConformedTo()`.